cli/run.ts

```diff
@@ -495,8 +495,17 @@ export async function run_generation_from_cli(desiredElmFile: string | null, opt
   let cwd = "./codegen"
 
   if (desiredElmFile != null) {
-    cwd = "."
-    elmFile = desiredElmFile
+    cwd = path.dirname(desiredElmFile)
+
+    while (!fs.existsSync(path.join(cwd, "elm.json"))) {
+      cwd = path.dirname(cwd)
```

Comment:
Nice!  Does dirname always return relative paths?

Basically want to be sure this doesn't infinitely hang 🤔

Reply:
Ah, I think it may hang if the `desidedElmFile` is an absolute path